libpromises/matching.c

```diff
@@ -292,7 +292,7 @@ void EscapeSpecialChars(const char *str, char *strEsc, int strEscSz, char *noEsc
                 break;
```

Comment:
A message here and  emptying the string would be nice - or something better than silently truncating.